### PR TITLE
Board and Deck classes

### PR DESCRIPTION
**Pull request by rseldon**
Create a class to represent the deck, which shuffles on initialization, and can deal one card.
Create a class for the Board, which can respond to a play, a discard, and a clue given

ToDo:
- [ ] unit tests for deck
- [ ] unit tests for board

### CODE REVIEW

File: test/test_game.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        bored_cat.play_card(white5)
+
+        self.assertEqual(bored_cat.num_clues, 1)
+
+        self.assertEqual(bored_cat.num_bombs, bombs)
+
+        for i in range(bombs):
+            while bored_cat.num_bombs != 0:
+                bored_cat.play_card(white3)
+                bored_cat.num_bombs -= 1
```
Comment (rseldon) [inline test/test_game.py]: I think `play_card` already decrements `num_bombs`, so the following line does it twice, so I think the `num_bombs` is ending at -3. I think line 76 can be removed

File: test/test_game.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        while card_drawn != yellow2:
+            card_drawn = default_deck.draw_card()
+
+        self.assertEqual(default_deck.card_counts[yellow2], 0)
+
+    def test_board(self):
+        # instantiate an instance of a Board
+        clues = 8
+        bombs = 3
+        bored_cat = Board(Deck(), max_clues=clues, bombs=3)
```
Comment (rseldon) [inline test/test_game.py]: it seems kind of weird that you create a variable for clues, then pass that in, but you create a variable for bombs and then pass in the constant 3 which happens to have the same value.

Comment (rseldon) [reply test/test_game.py]: I am informed that I wrote this, oops.

File: test/test_game.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+        with self.assertRaises(Exception):
+            bored_cat.give_clue()
+
+        white1 = Card(Color.WHITE, 1)
+        white2 = Card(Color.WHITE, 2)
+        white3 = Card(Color.WHITE, 3)
+        white4 = Card(Color.WHITE, 4)
+        white5 = Card(Color.WHITE, 5)
+        bored_cat.play_card(white1)
```
Comment (rseldon) [inline test/test_game.py]: This is fine, but if you want to make it more concise, you could write something like
```python
bored_cat.play_card(Card(Color.WHITE, 1))
```

**Review by rseldon — COMMENTED** [review]
Really Excellent work on these tests @jhenyie. I'm really impressed! Now it looks like we just need a few tests for discards, but looking really great so far.

File: firebreak/game.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        else:
+            self.num_bombs -= 1
+            self.discards[card] = self.discards.get(card, 0) + 1
+
+    def give_clue(self):
+        assert self.num_clues > 0, "Tried to give a clue when there were no available clue tokens!"
+        self.num_clues -= 1
+
+
+
```
Comment (rseldon) [inline firebreak/game.py]: too many blanks lines here (3 > 2)

File: test/test_game.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        self.assertEqual(bored_cat.num_clues, 0)
+
+        with self.assertRaises(Exception):
+            bored_cat.give_clue()
+
+        bored_cat.play_card(Card(Color.WHITE, 1))
+        bored_cat.play_card(Card(Color.WHITE, 2))
+        bored_cat.play_card(Card(Color.WHITE, 3))
+        bored_cat.play_card(Card(Color.WHITE, 4))
+        bored_cat.play_card(Card(Color.WHITE, 5))
```
Comment (rseldon) [inline test/test_game.py]: `bored_cat` is cute, but perhaps hard to read

**Review by jhenyie — APPROVED** [review]
seems good

**Pull request merged**